Hide supplemental-pack feature behind a feature flag

ydirson opened the pull request:
Not touching this branch as it is also the source of https://github.com/xenserver/host-installer/pull/28.

Merge is available in branch next as 4308d25fd03d282f074e05a53a66102c6110d580.

stormi commented:
Hi. I already considered this change before the release of XCP-ng 8.0, but Citrix did raise the amount of RAM in the installer for a reason, so reducing it again would simply shift the problem from some hardware to other.

I suppose the proper fix belongs to the kernel itself, which should probably not crash when there's 8Gb of RAM on this kind of hardware.

varnav commented:
Is there a bug filed about this in the upstream?

stormi commented:
Not by me, but I'd be interested to know if there's one and/or whether it's been fixed in the latest 4.19 LTS kernel.

stormi commented:
Let's continue the discussion to find a proper solution on the bug report https://github.com/xcp-ng/xcp/issues/206

ydirson committed:
Hide supplemental-pack feature behind a feature flag

The inclusion of supplemental packs occurs mid-way in the
installation.  For products with no use for them, it is a big
annoyance during interactive install.

Signed-off-by: Yann Dirson <yann.dirson@vates.fr>
